### PR TITLE
Add plans parsing to the apiserver

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

* The apiserver now parses plans from the `metadata.appcat.vshn.io/plans` annotation on the compositions.

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ EOF
 
 After the above steps just run the API server via IDE with the following arguments.
 ```
-apiserver --secure-port=9443 --kubeconfig=<your-home-path>/.kube/config --authentication-kubeconfig=<your-home-path>/.kube/config --authorization-kubeconfig=<your-home-path>/.kube/config
+apiserver --secure-port=9443 --kubeconfig=<your-home-path>/.kube/config --authentication-kubeconfig=<your-home-path>/.kube/config --authorization-kubeconfig=<your-home-path>/.kube/config --feature-gates=APIPriorityAndFairness=false
```
Comment (zugao) [inline README.md:141]: Great that you fixed this.

File: apis/appcat/v1/appcat_types.go [hunk 2]
```diff
@@ -50,9 +53,25 @@ type AppCatList struct {
 	Items []AppCat `json:"items"`
 }
 
-// AppCatSpec defines the desired state of AppCat
-// The desired state of AppCat is dynamically populated from composition's labels
-type AppCatSpec map[string]string
+// Details are fields that are dynamically parsed from the annotations on a composition.
+type Details map[string]string
```
Comment (zugao) [inline apis/appcat/v1/appcat_types.go:57]: Why the name of this field has been changed?

**Review by zugao — APPROVED** [review]
LGTM

File: apis/appcat/v1/appcat_types.go (hunk 2 above)
Comment (Kidswiss) [reply apis/appcat/v1/appcat_types.go:57]: Because there's no more spec field and calling it spec now is just wrong.

**Pull request merged**